src/planner/planner.c

```diff
@@ -444,8 +440,12 @@ timescaledb_planner(Query *parse, int cursor_opts, ParamListInfo bound_params)
 {
 	PlannedStmt *stmt;
 	ListCell *lc;
-	bool reset_fetcher_type = false;
-	bool reset_baserel_info = false;
+	/*
+	 * Volatile is needed because these are the local variables that are
+	 * modified between setjmp/longjmp calls.
+	 */
+	volatile bool reset_fetcher_type = false;
+	volatile bool reset_baserel_info = false;
```

Comment:
So that's what the `-Wclobbered` warning was about... I'll fix the other places separately.